fix: #231 adjust font size for How it works title to 32px

Sanchez86 opened the pull request:
- Changed title font size from 35px (h4 default) to 32px in HowItWorks component
- Added explicit fontSize property to override typography variant
- Resolves issue with bold descriptions display in home page How it works block

Sanchez86 committed:
fix: #231 adjust font size for How it works title to 32px

- Changed title font size from 35px (h4 default) to 32px in HowItWorks component
- Added explicit fontSize property to override typography variant
- Resolves issue with bold descriptions display in home page How it works block

AndiKot commented on src/containers/guest-home-page/how-it-works/HowItWorks.styles.ts:
```diff
@@ -10,5 +10,5 @@ export const styles = {
     flexDirection: { sm: 'row', xs: 'column' }
   },
-  title: { mb: '32px', typography: 'h3' }
+  title: { mb: '32px', typography: 'h4', fontSize: '32px' }
```
- You can use standard MUI spacing units the margin `mb: 4` (https://mui.com/system/spacing/)
- font-size for the mob - 32px and tab/desktop - 48px
- mb for the mob - 24px and tab/desktop - 32px

Sanchez86 replied:
Already done

Sanchez86 committed:
fix: update HowItWorks title styles with responsive design

- Use MUI standard spacing units (mb: { xs: 3, md: 4 })
- Add responsive font sizes: 32px mobile, 48px tablet/desktop
- Add responsive margins: 24px mobile, 32px tablet/desktop
- Improve mobile and desktop user experience

Addresses task #231

Sanchez86 committed:
trigger CI checks
